src/plugins/console/public/application/containers/editor/monaco/components/context_menu/context_menu.tsx

```diff
@@ -0,0 +1,293 @@
+}: Props) => {
+  // Get default language from local storage
+  const {
+    services: { storage, esHostService },
+  } = useServicesContext();
+  const defaultLanguage = storage.get(StorageKeys.DEFAULT_LANGUAGE, DEFAULT_LANGUAGE);
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [isRequestConverterLoading, setRequestConverterLoading] = useState(false);
+  const [isLanguageSelectorVisible, setLanguageSelectorVisibility] = useState(false);
+  const [currentLanguage, setCurrentLanguage] = useState(defaultLanguage);
+
+  const copyText = async (text: string) => {
+    if (window.navigator?.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+      return;
+    }
+    throw new Error('Could not copy to clipboard!');
+  };
+
+  // This function will convert all the selected requests to the language by
+  // calling convertRequestToLanguage for each request and then copy the data
+  // to clipboard.
+  const copyAs = async (language?: string) => {
+    // Get the language we want to convert the requests to
+    const withLanguage = (language || currentLanguage).toLowerCase();
+    // Get all the selected requests
+    const requests = await getRequests();
+
+    // Convert each request using convertRequestToLanguage and handle all promises
+    const results = await Promise.allSettled(
+      requests.map((request: any) =>
+        convertRequestToLanguage({
+          method: request.method,
+          path: request.url,
+          body: request.data,
+          language: withLanguage,
+          esHost: esHostService.getHost(),
+        })
+      )
+    );
+
+    // Aggregate data and log errors
+    let aggregatedData = '';
+    let hasErrors = false;
+
+    results.forEach((result) => {
+      if (result.status === 'fulfilled' && !result.value.error) {
+        aggregatedData += result.value.data + '\n';
+      } else {
+        hasErrors = true;
+      }
+    });
+
+    // If we dont have data and have errors, we should show an error toast saying
+    // that no requests could be converted
+    if (aggregatedData === '' && hasErrors) {
+      const error = new Error(`Failed to convert request to ${withLanguage}`);
+      notifications.toasts.addError(error, {
+        title: i18n.translate('console.consoleMenu.copyAsCurlFailedMessage', {
+          defaultMessage:
+            '{requestsCount, plural, one {Request} other {Requests}} could not be copied to clipboard',
+          values: { requestsCount: requests.length },
+        }),
+      });
+
+      return Promise.reject(error);
+    } else {
+      // If we have data and errors, we should show a warning toast saying that
+      // some requests could not be converted and copy the rest to clipboard
+      if (aggregatedData !== '' && hasErrors) {
+        notifications.toasts.addWarning({
+          title: i18n.translate('console.consoleMenu.copyAsCurlSuccessWithWarning', {
+            defaultMessage: 'Some requests could not be copied to clipboard',
```

Comment:
```suggestion
            defaultMessage: 'Some requests could not be copied to clipboard because they are invalid',
```

Reply:
Thanks for having a look, I've addressed this already with https://github.com/elastic/kibana/pull/189976/commits/cb01b57f815afffd37e2ba92162ff79e311e2464 🚀